stl/inc/flat_map

```diff
@@ -0,0 +1,873 @@
+    iterator emplace_hint(const_iterator _Position, _Args_t&&... _Args)
+        requires _STD
+    is_constructible_v<value_type, _Args_t...> {
+        value_type _Val(_STD forward<_Args_t>(_Args)...);
+
+        return _Emplace_hint<false, decltype(_Val.first)&&, decltype(_Val.second)&&>(
+            _Position, _STD move(_Val.first), _STD move(_Val.second));
+    }
+
+    template <class _V>
+    _STD pair<iterator, bool> insert(_V&& _X)
+        requires (same_as<_STD remove_cvref_t<_V &&>, value_type> || constructible_from<value_type, _V &&>)
+    {
+        return emplace(_STD forward<_V>(_X));
+    }
+
+    template <class _V>
+    iterator insert(const_iterator _Position, _V&& _X)
+        requires (same_as<_STD remove_cvref_t<_V &&>, value_type> || constructible_from<value_type, _V &&>)
+    {
+        return emplace_hint(_Position, _STD forward<_V>(_X));
+    }
+
+    template <class _InputIterator>
+        requires _Is_iterator_v<_InputIterator>
+    void insert(_InputIterator _First, _InputIterator _Last) {
+        _Insert_range<_InputIterator, true, true>(_First, _Last);
+    }
+
+    template <class _InputIterator>
+        requires _Is_iterator_v<_InputIterator>
+    void insert(sorted_unique_t, _InputIterator _First, _InputIterator _Last) {
+        _Insert_range<_InputIterator, false, true>(_First, _Last);
+    }
+
+    template <_Container_compatible_range<value_type> R>
+    void insert_range(R&& _Range) {
+        insert(_RANGES begin(_Range), _RANGES end(_Range));
+    }
+
+    void insert(initializer_list<value_type> _I) {
+        insert(_I.begin(), _I.end());
+    }
+
+    void insert(sorted_unique_t _S, initializer_list<value_type> _I) {
+        insert(_S, _I.begin(), _I.end());
+    }
+
+    template <class _Key_constructible_t, class... _Args_t>
+    _STD pair<iterator, bool> try_emplace(_Key_constructible_t&& _Key_constructible, _Args_t&&... _Args)
+        requires constructible_from<mapped_type, _Args_t&&...>
+              && (same_as<_STD remove_cvref_t<_Key_constructible_t &&>, key_type>
+                  || (constructible_from<key_type, _Key_constructible_t&&> && _Compare_is_transparent<key_compare>
+                      && !convertible_to<_Key_constructible_t&&, const_iterator>
+                      && !convertible_to<_Key_constructible_t&&, iterator>) )
+    {
+        auto _Key_It = _STD lower_bound(
+            _Data.keys.begin(), _Data.keys.end(), _STD forward<_Key_constructible_t>(_Key_constructible), _Key_compare);
+        if (_Key_It != _Data.keys.end()
+            && _Key_equal(*_Key_It, _STD forward<_Key_constructible_t>(_Key_constructible))) {
+            // Already exists
+            return _STD make_pair(begin() + _STD distance(_Data.keys.begin(), _Key_It), false);
+        } else {
+            // Need to insert
+            value_type _Val = std::make_pair(key_type(_STD forward<_Key_constructible_t>(_Key_constructible)),
+                mapped_type(_STD forward<_Args_t>(_Args)...));

```

Comment:
Moving can be reduced by piecewise construction.
```suggestion
            value_type _Val{piecewise_construct,
                _STD forward_as_tuple(_STD forward<_Key_constructible_t>(_Key_constructible)),
                _STD forward_as_tuple(_STD forward<_Args_t>(_Args)...)};
```

Reply:
Done.